### PR TITLE
use scoped interval

### PR DESCRIPTION
**Pull request by levnach**
Signed-off-by: Lev Nachmanson <levnach@hotmail.com>

### CODE REVIEW

File: src/math/lp/nla_intervals.cpp [hunk 1]
```diff
@@ -302,72 +304,68 @@ bool intervals::interval_from_term(const nex& e, interval& i) {
...
 template <e_with_deps wd>
-intervals::interval intervals::interval_of_sum_no_term(const nex_sum& e) {
-    const nex* inf_e = get_inf_interval_child(e);
-    if (inf_e) {
-        return interval();
-    }
-    interval a = interval_of_expr<wd>(e[0], 1);
+void intervals::interval_of_sum_no_term(const nex_sum& e, scoped_dep_interval & sdi) {
+    if (has_inf_interval(e))
+        return;
```
Comment (NikolajBjorner) [inline src/math/lp/nla_intervals.cpp:309]: would be good to set sdi to interval() explicitly.

File: src/math/lp/nla_intervals.cpp [hunk 2]
```diff
@@ -302,72 +304,68 @@ bool intervals::interval_from_term(const nex& e, interval& i) {
...
 
-    interval a;
     m_dep_intervals.set_interval_for_scalar(a, e.coeff());
     TRACE("nla_intervals_details", tout << "a = "; display(tout, a); );
     for (const auto& ep : e) {
-        interval b = interval_of_expr<wd>(ep.e(), ep.pow());
+        scoped_dep_interval b(get_dep_intervals());
+        interval_of_expr<wd>(ep.e(), ep.pow(), b);
         TRACE("nla_intervals_details", tout << "ep = " << ep << ", "; display(tout, b); );
         interval c;
```
Comment (NikolajBjorner) [inline src/math/lp/nla_intervals.cpp:370]: make all intervals scoped, also c?

File: src/math/lp/nla_intervals.cpp (hunk 1 above)
Comment (levnach) [reply src/math/lp/nla_intervals.cpp:309]: I can add an assert that makes sure the bounds are infinite.

File: src/math/lp/nla_intervals.cpp (hunk 2 above)
Comment (levnach) [reply src/math/lp/nla_intervals.cpp:370]: ok

**Pull request merged**